src/Conversion/ONNXToTOSA/Tensor/Resize.cpp

```diff
@@ -0,0 +1,314 @@
+        : numerator(numerator), denominator(denominator), offset(offset),
+          border(border){};
+    int64_t numerator, denominator, offset, border;
+  };
+
+  struct FractionNumber {
+    FractionNumber(int64_t numerator, int64_t denominator)
+        : numerator(numerator), denominator(denominator){};
+    FractionNumber(float number) {
+      double integral = std::floor(number);
+      double frac = number - integral;
+
+      const long precision = 1000000000; // This is the accuracy.
+
+      long gcd_ = std::gcd((int)round(frac * precision), precision);
+
+      long denominator = precision / gcd_;
+      long numerator = round(frac * precision) / gcd_;
+
+      this->numerator = numerator + denominator * integral;
+      this->denominator = denominator;
+    }
+    int64_t numerator;
+    int64_t denominator;
+  };
+
+  /// ## coordinateTransformationMode ##
+  /// TOSA uses the formula ix = (ox * scale_x_d + offset_x) / scale_x_n
+  /// to find the input coordinates. In order to lower ONNX one needs to
+  /// express the modes in this context. Border is only used to check if certain
+  /// conditions in the dimensions are met, but has no actual use in calculating
+  /// something. It is probably an error in the TOSA specification. In order to
+  /// fulfill the conditions, border is always
+  /// border = d * (output - 1) - n *(input - 1) + offset
+  ///
+  /// ### half_pixel ###
+  /// ONNX formula: ix = (ox + 0.5) / scale - 0.5
+  /// gcd = greatest common divisor
+  /// To meet TOSA requirements:
+  /// - scale_x_d = input_size * 2 / gcd
+  /// - scale_x_n = output_size * 2 / gcd
+  /// - offset_x =  (scale_x_d - scale_x_n) / 2
+  ///
+  /// ### pytorch_half_pixel ###
+  /// Same as half_pixel, but if output == 1:
+  /// - scale_x_d = scale_x_n = 1
+  /// - offset = -1
+  ///
+  /// ### half_pixel_symmetric ###
+  /// NOT SUPPORTED BY TOSA, BECAUSE OF INT OFFSET
+  /// Same as half_pixel, but with another offset
+  /// - offset_x += symmetric_offset_x
+  ///
+  /// ### align_corners ###
+  /// - scale_x_d = (input_size - 1)
+  /// - scale_x_n = (output_size - 1)
+  /// - offset = 0
+  ///
+  /// ### asymmetric ###
+  /// - scale_x_d = input_size
+  /// - scale_x_n = output_size
+  /// - offset = 0
+  ///
+  /// ## nearest_mode ##
+  /// If mode == nearest, then ONNX can set the nearest_mode attribute.
+  /// The standard case for TOSA of round_half_up. Support for floor can
+  /// be achieved with:
+  /// offset_x -= n/2
+
+  LogicalResult matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const final {
+    auto resizeOp = llvm::cast<ONNXResizeOp>(op);
+    Location loc = op->getLoc();
+    OpAdaptor adaptor(operands, op->getAttrDictionary());
+
+    TosaBuilder tosaBuilder(rewriter, loc);
+
+    Value input = adaptor.getX();
+    auto inputType = input.getType().dyn_cast<RankedTensorType>();
+
+    auto resultType =
```

Comment:
I think we should also check that ONNX is using the supported types in the TOSA specification. `in_t` only supports `i8, i16, fp16, bf16 and fp32` and it seems to depend on the mode. 

https://www.mlplatform.org/tosa/tosa_spec.html#_resize

Reply:
because onnx also supports unsigned in resize.

Reply:
Yeah so basically TOSA allows everything...